acapy_agent/resolver/routes.py

```diff
@@ -23,7 +24,10 @@ class W3cDID(validate.Regexp):
     """Validate value against w3c DID."""
 
     EXAMPLE = "did:ted:WgWxqztrNooG92RXvxSTWv"
-    PATTERN = DID_PATTERN
+    # Did or DidUrl regex
+    PATTERN = re.compile(
+        "^did:([a-z0-9]+):((?:[a-zA-Z0-9._%-]*:)*[a-zA-Z0-9._%-]+)(#\w+)?$"
+    )
```

Comment:
Changes the pattern to allow did url pattern. Couldn't figure out how to do it with the pydid library.

Reply:
I'm confused why we're doing this. A DID is distinct from a DID URL?

Reply:
I suspect it is because a DID resolved to a DIDDoc, and a DID URL, resolves to whatever the resource is pointed to by the URL (AnonCreds object, OCA bundle, status list VC, etc). The DID still needs to be resolved (perhaps from a cache) before getting the resource, but the resolution result object will be different.

Reply:
For clarity (forgive the pedantism), we resolve DIDs but dereference DID URLs. I think it's valuable to keep the two operations distinct since they're distinct in the DID Core spec. Resolving a DID results in a DID Document. Dereferencing a DID URL results in a "resource." So rather than altering the existing resolve endpoint to accept a DID URL, I'd recommend a separate endpoint for that.

Reply:
So the resolver endpoint should fail for something like `did:tdw:QmZiKXwQVfyZVuvCsuHpQh4arSUpEmeVVRvSfv3uiEycSr:example.com%3A5000#whois`, and pass for `did:tdw:QmZiKXwQVfyZVuvCsuHpQh4arSUpEmeVVRvSfv3uiEycSr:example.com%3A5000`? I guess I was a bit confused because the did_tdw library resolves `did:tdw:QmZiKXwQVfyZVuvCsuHpQh4arSUpEmeVVRvSfv3uiEycSr:example.com%3A5000#whois` 

I think updating this to just resolve dids shouldn't be a problem. I can't remember if I was getting validation errors with a valid did:tdw without the `#` or not. Will refactor it back and check.

Reply:
The `resolve_did` method which is used here does not perform dereferencing (it returns a `ResolutionResult` not a `DereferencingResult`). It does not currently reject a DID URL with a path or extra parameters, although I might make that change to avoid misuse.

Reply:
@jamshale is using `#whois` in the examples above, but I think (I hope) they should be `/whois`.  To me, `<did>#` should be for getting a fragment of the DIDDoc.  The `<did>/` is for getting a resource separate from the DIDDoc. And `whois` is an external resource.

That said, I have wondered why the `did:tdw` spec says the implicit service is `#whois`. I wrote what I (think I) was told.  Is that right to get the `/` behaviour?

Reply:
`#whois` does indeed dereference to the service definition, not the whois document itself. I believe `/whois` should be translated to `?service=whois` by the resolver and dereferenced that way.